Client/src/main/java/com/minecrafttas/tasbattle/SpectatorManager.java

```diff
@@ -0,0 +1,232 @@
+
+/**
+ * Adds custom spectating modes. This is clientside only
+ * @author Scribble
+ *
+ */
+public class SpectatorManager {
+
+	@Getter
+	private static SpectatorManager instance;
+
+	/**
+	 * The entity that 
+	 */
+	private Entity spectatedEntity;
+
+	/**
+	 * The current specatting mode of the manager
+	 * 
+	 * @see SpectatorMode
+	 */
+	private SpectatorMode mode = SpectatorMode.None;
+	
+	public SpectatorManager() {
+		instance = this;
+	}
+
+	/**
+	 * Toggles between the {@link SpectatorMode#FixedAngle} mode and {@link SpectatorMode#None}
+	 */
+	public void cycleSpectate() {
+		List<SpectatorMode> modes = SpectatorMode.valueList();
+		setMode(nextObject(modes, mode));
+		TASBattle.LOGGER.info("Cycling spectate to {}", mode);
+	}
+	
+	/**
+	 * Sets the {@link #mode} and runs initializing code for that mode.
+	 * @param mode The {@link SpectatorMode} that should be set to {@link #mode}
+	 */
+	public void setMode(SpectatorMode mode) {
+		if(mode == SpectatorMode.None) {
+			spectatedEntity = null;
+			this.mode = mode;
+			return;
+		}
+		
+		if (spectatedEntity == null) {
+			spectatedEntity = getNearestSpectatedEntity();
+		}
+		if (spectatedEntity != null) {
+			this.mode = mode;
+		}
+	}
+
+	private Entity getNearestSpectatedEntity() {
+		Entity nearestEntity = getNearestPlayer(100); // Get the nearest player
+		
+		if(nearestEntity == null && TASBattle.isDevEnvironment()) {	// If indev, fall back to the nearest entity if no player was found
+			nearestEntity = getNearestEntity(100);
+		}
+		
+		if(nearestEntity != null) {
+			return nearestEntity;
+		}
+		return null;
+	}
+	
+	/**
+	 * Selects the next player/entity in range as the {@link #spectatedEntity}
+	 */
+	public void spectateNextPlayer() {
+		Minecraft mc = Minecraft.getInstance();
+		List<AbstractClientPlayer> playerList = mc.level.players();
+		playerList.removeIf(player -> player.isSpectator()); // Remove spectators from the list
+
+		if (!playerList.isEmpty()) {
+			spectatedEntity = nextObject(playerList, spectatedEntity);
+		} else if (TASBattle.isDevEnvironment()) {
+			List<Entity> entitiesList = mc.level.getEntities(mc.player, mc.player.getBoundingBox().inflate(100));
+			spectatedEntity = nextObject(entitiesList, spectatedEntity);
+		}
+	}
+
+	/**
+	 * Selects the previous player/entity in range as the {@link #spectatedEntity}
+	 */
+	public void spectatePreviousPlayer() {
+		Minecraft mc = Minecraft.getInstance();
+		List<AbstractClientPlayer> playerList = mc.level.players();
+		playerList.removeIf(player -> player.isSpectator()); // Remove spectators from the list
+
+		if (!playerList.isEmpty()) {
+			spectatedEntity = previousObject(playerList, spectatedEntity);
+		} else if (TASBattle.isDevEnvironment()) {
+			List<Entity> entities = mc.level.getEntities(mc.player, mc.player.getBoundingBox().inflate(100));
+			spectatedEntity = previousObject(entities, spectatedEntity);
+		}
+	}
+
+	private Double anglePitch;
+	private Double angleYaw;
+	
```

Comment:
why in the world is this uppercase Double

Reply:
so it can be null